docs/sapling.md

```diff
@@ -0,0 +1,277 @@
+
+Refers to the following link for more information: [InMemoryViewingKey](./sapling_in_memory_viewing_key.md)
+
+**Address**
+
+Addresses are used to receive tokens. They are derived from the viewing key.
+
+Here is an example on how to retrieve addresses: [InMemoryViewingKey](./sapling_in_memory_viewing_key.md#how-to-retrieve-payment-addresses-from-the-viewing-key)
+
+
+# Sapling toolkit
+
+The `@taquito/sapling` package provides a `SaplingToolkit` class that surfaces all of the sapling capabilities allowing to read from a sapling state and prepare transactions. 
+
+The constructor of the `SaplingToolkit` takes the following properties:
+- an instance of `InMemorySpendingKey` as the spending key is needed to prepare and sign transactions that spend tokens
+- the second parameter is an object containing:
+  - the address of the sapling contract (string)
+  - the size of the memo of the corresponding sapling contract (number)
+  - an optional sapling id that must be specified if the contract contains more than one sapling state.
+- an instance of a class implementing the `TzReadProvider` interface, which allows getting data from the blockchain
+- it is possible to specify a different packer than the `MichelCodecPacker`, which is used by default
+
+Here is an example of how to instantiate a `SaplingToolkit`:
+
+```ts
+import { TezosToolkit } from '@taquito/taquito';
+import { SaplingToolkit } from '@taquito/sapling';
+
+const tezos = new TezosToolkit('https://jakartanet.ecadinfra.com/');
+const saplingContract = await tezos.contract.at('KT1UYwMR6Q6LZnwQEi77DSBrAjKT1tEJb245');
+
+const inMemorySpendingKey = await InMemorySpendingKey.fromMnemonic('YOUR_MNEMONIC');
+
+const saplingToolkit = new SaplingToolkit(
+    inMemorySpendingKey, 
+    { contractAddress: saplingContract.address, memoSize: 8 }, 
+    tezos.getFactory(RpcReadAdapter)()
+)
+```
+
+## How to retrieve my balance in the Sapling shielded pool?
+
+When calling the `getSaplingTransactionViewer` method of the `SaplingToolkit` class, an instance of the `SaplingTransactionViewer` class is returned. The `SaplingTransactionViewer` class allows retrieving and decrypting sapling transactions for the specified viewing key and calculating the unspent balance. 
+
+For each entry in the shielded pool, the `SaplingTransactionViewer` class will try to decrypt them using the viewing key as if it were the receiver. If a ciphertext is successfully decrypted, the configured account was the receiver of the output. The `SaplingTransactionViewer` will find which inputs were not spent by computing their nullifier. If an input is spent, its nullifier will be in the sapling state. If the nullifier is not present, thus the input has not been spent, and its value will be considered in the calculated balance.
```

Comment:
"If the nullifier is not present, thus the input has not been spent, and its value will be considered in the calculated balance."
 -> "If the nullifier is not present, the input has not been spent, and its value will be considered in the calculated balance.

Reply:
thanks